src/ProcessSlave.php

```diff
@@ -188,24 +187,24 @@ function ($item) {
 
         $this->inShutdown = true;
 
-        if ($this->loop) {
-            $this->sendCurrentFiles();
-            $this->loop->tick();
-        }
-
         if ($this->controller && $this->controller->isWritable()) {
             $this->controller->close();
         }
         if ($this->server) {
             @$this->server->close();
         }
+
         if ($this->loop) {
-            $this->sendCurrentFiles();
-            $this->loop->tick();
-            @$this->loop->stop();
+            $this->loop->futureTick(function () {
+                // watch source files with potentially fatal error for changes
+                $this->sendCurrentFiles();
+                $this->loop->stop();
+                exit;
+            });
+        }
```

Comment:
`} else {`

Reply:
I'd go further and say use early returning statements.

so, we have

```php
if (!this->loop) {
    exit;
}

$this->loop->futureTick(function () {
    $this->loop->tick();
    // watch source files with potentially fatal error for changes
    $this->sendCurrentFiles();
    $this->loop->stop();
    exit;
});

```

Reply:
Updated accordingly. `tick` is no longer part of the API.